### PR TITLE
Implementation of MtE Pade operator

### PR DESCRIPTION
**Pull request by mscroggs**
This is the same as #152 but with style checks fixed so that the tests will run

### CODE REVIEW

File: bempp/api/operators/boundary/maxwell.py [hunk 1]
```diff
@@ -289,3 +289,142 @@ def _multitrace_operator_impl(
...
+        [wavenumber, npade, theta, damped_wavenumber],
+        device_interface,
+        precision,
+    )
+
+
+def Sum(A, B, inpt, inner_coeff, outer_coeff):
+    """Return a sum.
+
+    TODO: document this.
```
Comment (mscroggs) [inline bempp/api/operators/boundary/maxwell.py:413]: We should document this function

File: bempp/api/operators/boundary/sparse.py [hunk 2]
```diff
@@ -58,6 +108,55 @@ def multitrace_identity(
...
+    LT = grad_identity(domains_[1], ranges_[0], dual_to_ranges_[0])
+    L = LT._transpose(LT._domain)
+    return IP, IC, N, LT, L
+
+
+def mte_lambda_1i(mte_operators, beta, kappa):
+    """
+    Crate the MTE lambda 1i operator.
+
+    TODO: document this
```
Comment (mscroggs) [inline bempp/api/operators/boundary/sparse.py:141]: We should document this function

File: bempp/api/operators/boundary/sparse.py [hunk 3]
```diff
@@ -58,6 +108,55 @@ def multitrace_identity(
...
+    IP, IC, N, LT, L = mte_operators
+    return InverseSparseDiscreteBoundaryOperator(
+        bmat([[(IC - beta * N).weak_form().to_sparse(), (beta * LT).weak_form().to_sparse()], [L.weak_form().to_sparse(), kappa ** 2 * IP.weak_form().to_sparse()]], 'csc'))
+
+
+def mte_lambda_2(mte_operators):
+    """
+    Crate the MTE lambda 2 operator.
+
+    TODO: document this
```
Comment (mscroggs) [inline bempp/api/operators/boundary/sparse.py:154]: ... and this one

File: bempp/api/operators/boundary/sparse.py (hunk 2 above)
Comment (ignacia-fp) [reply bempp/api/operators/boundary/sparse.py:141]: Done

File: bempp/api/operators/boundary/maxwell.py (hunk 1 above)
Comment (ignacia-fp) [reply bempp/api/operators/boundary/maxwell.py:413]: Done

File: bempp/api/operators/boundary/sparse.py (hunk 3 above)
Comment (ignacia-fp) [reply bempp/api/operators/boundary/sparse.py:154]: Done

**Review by ignacia-fp — COMMENTED** [review]
some comments solved and changed in main included

**Review by mscroggs — COMMENTED** [review]
I've added comments from the in person meeting

**Pull request merged**